ehr/src/org/labkey/ehr/table/DefaultEHRCustomizer.java

```diff
@@ -638,18 +615,36 @@ private void customizeStudyData(AbstractTableInfo ti)
         customizeButtonBar(ti);
     }
 
+    private void expandRemarksColumn(final AbstractTableInfo ti)
+    {
+        Dataset dataset = ((DatasetTable) ti).getDataset();
+        Domain domain = dataset.getDomain();
+
+        // Hack the metadata for the length of the remark column, which has
+        // been altered in the DB so that it doesn't match with the standard
+        // "remark" property. See ONPRC ticket 33848 and EHRManager.ensureDatasetPropertyDescriptors()
+        TableInfo realTable = StorageProvisioner.getSchemaTableInfo(domain);
+        MutableColumnInfo remarkCol = (MutableColumnInfo) realTable.getColumn("remark");
+        remarkCol.setLocked(false);
+        remarkCol.setScale(1000000);
+        remarkCol.setLocked(true);
+    }
```

Comment:
@labkey-jeckels This is throwing below error during study import, since looks like remark column is "locked" and cannot be unlocked/modified this way. For ONPRC's use case, can we just add the scale for 'remark' column in datasets_metadata.xml?

java.lang.IllegalStateException: Can't unlock a ColumnInfo: remark
    at org.labkey.api.data.BaseColumnInfo.setLocked(BaseColumnInfo.java:2179)
    at org.labkey.ehr.table.DefaultEHRCustomizer.expandRemarksColumn(DefaultEHRCustomizer.java:630)
    at org.labkey.ehr.table.DefaultEHRCustomizer.customizeEncountersTable(DefaultEHRCustomizer.java:644)
    at org.labkey.ehr.table.DefaultEHRCustomizer.customizeDataset(DefaultEHRCustomizer.java:463)
    at org.labkey.ehr.table.DefaultEHRCustomizer.customize(DefaultEHRCustomizer.java:123)

Reply:
In my testing we can just get rid of this completely. The length validation code now allows the length through based on the underlying column in the table.

Reply:
done